### PR TITLE
Add a link to be able to visit the PWA

### LINKED ISSUE #67: Add a link to be able to visit the PWA from the detail page.
Add a link to be able to visit the PWA from the detail page.

Needs #66

### PR DESCRIPTION
**Pull request by juliantoledo**
Fixes #67 and #66

### CODE REVIEW

**Comment by juliantoledo** [comment]
@ithinkihaveacat PTAL

File: models/pwa.js [hunk 1]
```diff
@@ -31,6 +32,14 @@ const E_MANIFEST_ERROR = exports.E_MANIFEST_ERROR = 2;
...
   pwa.name = manifest.name;
   pwa.startUrl = manifest.start_url || '';
+
+  // We need an absolut URL to the PWA (Manifest can include both types)
+  if (manifest.start_url.indexOf('http://') === 0 || manifest.start_url.indexOf('https://') === 0) {
+    pwa.absolutStartUrl = manifest.start_url || '';
+  } else {
+    pwa.absolutStartUrl = uri(manifest.start_url).absoluteTo(manifest.url).toString() || '';
+  }
+
```
Comment (ithinkihaveacat) [inline models/pwa.js]: I don't think you need to check to see if `start_url` is relative or not: I think you can just throw whatever we get from the manifest at `urijs` and let it figure it out:

```js
Uri("/?homescreen=1").absoluteTo("https://example.com/manifest.json").toString()
// -> https://example.com/?homescreen=1
Uri("https://foo.com/something/?homescreen=1").absoluteTo("https://example.com/manifest.json").toString()
// -> https://foo.com/something/?homescreen=1
```

Comment (juliantoledo) [inline models/pwa.js]: Good catch, I will try it, I was doing something different before using Uri.

**Comment by juliantoledo** [comment]
@ithinkihaveacat I've fixed the URI.

File: .jshintrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "esnext": true,
+  "node": true
+}
```
Comment (ithinkihaveacat) [inline .jshintrc]: What's the purpose of running jshint as well as eslint? Currently the CI only runs eslint (using rules from `.eslintrc.json`).

**Comment by ithinkihaveacat** [comment]
LGTM

File: models/pwa.js [hunk 3]
```diff
@@ -31,6 +32,7 @@ const E_MANIFEST_ERROR = exports.E_MANIFEST_ERROR = 2;
 function mergeManifest(pwa, manifest) {
   pwa.name = manifest.name;
   pwa.startUrl = manifest.start_url || '';
+  pwa.absolutStartUrl = uri(manifest.start_url).absoluteTo(manifest.url).toString() || '';
```
Comment (ithinkihaveacat) [inline models/pwa.js]: s/absolutStartUrl/absoluteStartUrl/?

File: views/pwas/view.hbs [hunk 4]
```diff
@@ -42,14 +42,16 @@
...
         </div>
 
         <div class="pwabox-body">
-          <h4>{{pwa.name}}</h4>
-          <h4><a href="{{pwa.manifestUrl}}">{{pwa.manifestUrl}}</a></h4>
-          <p>{{pwa.description}}</p>
-          <p>{{startUrl}}</p>
+          <h3>{{pwa.name}}</h4>
+          <p>{{pwa.manifestData.description}}</p>
+          {{#if pwa.absolutStartUrl}}
```
Comment (ithinkihaveacat) [inline views/pwas/view.hbs]: Also here?

File: .jshintrc (hunk 2 above)
Comment (juliantoledo) [inline .jshintrc]: It's for the jshint sublime plugin. I can remove it.

Comment (ithinkihaveacat) [inline .jshintrc]: Ah, is there an eslint one that can be used instead? I think it'd be better to use the same rules/linter everywhere if possible.

**Pull request merged**